flexmeasures/data/models/generic_assets.py

```diff
@@ -578,16 +605,24 @@ def create_generic_asset(generic_asset_type: str, **kwargs) -> GenericAsset:
     ).one_or_none()
     if generic_asset_type is None:
         raise ValueError(f"Cannot find GenericAssetType {asset_type_name} in database.")
-    new_generic_asset = GenericAsset(
+
+    new_generic_asset = get_or_create_generic_asset(
         name=kwargs["name"],
         generic_asset_type_id=generic_asset_type.id,
         attributes=kwargs["attributes"] if "attributes" in kwargs else {},
     )
+
+    # GenericAsset(
+    #     name=kwargs["name"],
+    #     generic_asset_type_id=generic_asset_type.id,
+    #     attributes=kwargs["attributes"] if "attributes" in kwargs else {},
+    # )
+
     for arg in ("latitude", "longitude", "account_id"):
         if arg in kwargs:
             setattr(new_generic_asset, arg, kwargs[arg])
```

Comment:
This now happens after adding the GenericAsset to the database, and after flushing. Do these asset properties still end up in the database? Or do we end up with a virtual (no lat/lng) public (no account_id) asset?

Reply:
Good point! 

In the latest commit, I've get these arguments passed to the `get_or_create_generic_asset`.